website/content/docs/security/acl/acl-policy-examples.mdx

```diff
@@ -0,0 +1,591 @@
+
+```hcl
+service "example-service" {
+  policy = "write"
+  intention = "write"
+}
+```
+
+</Tab>
+
+<Tab heading="ENT">
+
+The following policy grants a service owner permission to control intentions
+for incoming connections and requests to the `example-service` service
+in the `prod` partition in the `example-team` namespace.
+
+```hcl
+partition "prod" {
+  namespace "example-team" {
+    service "example-service" {
+      policy = "write"
+      intention = "write"
+    }
+  }
+}
+```
+
+</Tab>
+
+<!-- Service Owners -->
+</Tabs> 
+
+</Tab>
+
+
+<Tab heading="View Intentions">
+
+The following policies allow viewing all intentions.
+
+<Tabs>
+
+<Tab heading="OSS">
+
+```hcl
+service "" {
+  policy = "read"
+  intention = "read"
+}
+```
+
+</Tab>
+
+<Tab heading="ENT">
+
+```hcl
+partition_prefix "" {
+  namespace_prefix "" {
+    service "" {
+      policy = "read"
+      intention = "read"
+    }
+  }
+}
+```
+
+</Tab>
+
+</Tabs>
+
+</Tab>
+
+</Tabs>
+
+
+## Enable the Snapshot Agent to Run on a Specific Node
+
+The `consul snapshot agent` command starts a process that takes snapshots of the state of the Consul servers and either saves them locally or pushes them to a remote storage service. Refer to [Consul Snapshot Agent](/consul/commands/snapshot/agent) for additional information.
+
+In the following example, the ACL policy enables the snapshot agent to run on a node named `server-1234`.
+
+```hcl
+# Required to read and snapshot ACL data
+acl = "write"
+# Allow the snapshot agent to create the key consul-snapshot/lock which will
+# serve as a leader election lock when multiple snapshot agents are running in
+# an environment
+key "consul-snapshot/lock" {
+  policy = "write"
+}
+# Allow the snapshot agent to create sessions on the specified node
+session "server-1234" {
+  policy = "write"
+}
+# Allow the snapshot agent to register itself into the catalog
+service "consul-snapshot" {
+  policy = "write"
+}
+```
+
+## Enable Vault to Access the Consul Storage Backend
+
+If you are using [Vault](/vault/docs) to manage secrets in your infrastructure, you can configure Vault to use Consul's key/value (KV) store as backend storage to persist Vault's data. Refer to the [Consul KV documentation](/consul/docs/dynamic-app-config/kv) and the [Vault storage documentation](/vault/docs/configuration/storage) for additional information.
+
+In the following example, the ACL policy enables Vault to register as a service
+and provides access to the `vault/` path in Consul's KV store.
+
+```hcl
+# Provide KV visibility to all agents.
+agent_prefix "" {
+  policy = "read"
+}
+# Enable resources prefixed with 'vault/' to write to the KV
+key_prefix "vault/" {
+  policy = "write"
+}
+# Enable the vault service to write to the KV
+service "vault" {
+  policy = "write"
+}
+# Enable the agent to initialize a new session.
+session_prefix "" {
+  policy = "write"
+}
+```
+
+## Enable Consul External Service Monitor (ESM)
```

Comment:
Pinging @eikenb for review of the ACL policies for Consul ESM, for OSS and ENT 🙂 I tested the ones here and they worked for me locally. 

I was not able to get Consul ESM to run successfully in a non-default partition, so I included the following line:

> Consul ESM does not currently support non-default partitions.

Is that accurate?